src/rcv/unicore.c

```diff
@@ -0,0 +1,918 @@
+	return gpst2time(week, tow);
+}
+
+/* check code priority and return freq-index ---------------------------------*/
+static int checkpri(const char* opt, int sys, int code, int idx)
+{
+	int nex = NEXOBS;
+
+	if (sys == SYS_GPS) {
+		if (strstr(opt, "-GL1L") && idx == 0) return (code == CODE_L1L) ? 0 : -1;
+		if (strstr(opt, "-GL2S") && idx == 1) return (code == CODE_L2X) ? 1 : -1;
+		if (strstr(opt, "-GL2P") && idx == 1) return (code == CODE_L2P) ? 1 : -1;
+		if (code == CODE_L1L) return (nex < 1) ? -1 : NFREQ;
+		if (code == CODE_L2S) return (nex < 2) ? -1 : NFREQ + 1;
+		if (code == CODE_L2P) return (nex < 3) ? -1 : NFREQ + 2;
+	}
+	else if (sys == SYS_GLO) {
+		if (strstr(opt, "-RL2C") && idx == 1) return (code == CODE_L2C) ? 1 : -1;
+		if (code == CODE_L2C) return (nex < 1) ? -1 : NFREQ;
+	}
+	else if (sys == SYS_GAL) {
+		if (strstr(opt, "-EL6B") && idx == 3) return (code == CODE_L6B) ? 3 : -1;
+		if (code == CODE_L6B) return (nex < 2) ? -1 : NFREQ;
+	}
+	else if (sys == SYS_QZS) {
+		if (strstr(opt, "-JL1L") && idx == 0) return (code == CODE_L1L) ? 0 : -1;
+		if (strstr(opt, "-JL1Z") && idx == 0) return (code == CODE_L1Z) ? 0 : -1;
+		if (code == CODE_L1L) return (nex < 1) ? -1 : NFREQ;
+		if (code == CODE_L1Z) return (nex < 2) ? -1 : NFREQ + 1;
+	}
+	else if (sys == SYS_CMP) {
+		if (strstr(opt, "-CL1P") && idx == 0) return (code == CODE_L1P) ? 0 : -1;
+		if (strstr(opt, "-CL7D") && idx == 0) return (code == CODE_L7D) ? 0 : -1;
+		if (code == CODE_L1P) return (nex < 1) ? -1 : NFREQ;
+		if (code == CODE_L7D) return (nex < 2) ? -1 : NFREQ + 1;
+	}
+	return idx < NFREQ ? idx : -1;
+}
+
+/* signal type to obs code ---------------------------------------------------*/
+static int sig2code(int sys, int sigtype)
+{
+	if (sys == SYS_GPS) {
+		switch (sigtype) {
+		case 0: return CODE_L1C;  // L1C/A 
+		case 3: return CODE_L1L;  // L1C/A 
+		case 5: return CODE_L2P;  // L2P
+		case 6: return CODE_L5I;  // L5 Data 
+		case 9: return CODE_L2W;  // L2P(Y),semi-codeless 
+		case 11: return CODE_L1S; // L2P(Y),semi-codeless 
+		case 14: return CODE_L5Q; // L5 Pilot 
+		case 16: return CODE_L1L; // L1C(P)
+		case 17: return CODE_L2L; // L2C(L)
+		default: return 0;
+		}
+	}
+	else if (sys == SYS_GLO) {
+		switch (sigtype) {
+		case  0: return CODE_L1C; // L1C/A
+		case  5: return CODE_L2C; // L2C/A
+		case  6: return CODE_L3I; // G3I
+		case  7: return CODE_L3Q; // G3Q
+		default: return 0;
+		}
+	}
+	else if (sys == SYS_GAL) {
+		switch (sigtype) {
+		case  1: return CODE_L1A; // E1B
+		case  2: return CODE_L1C; // E1C
+		case 12: return CODE_L5Q; // E5A Pilot
+		case 17: return CODE_L7Q; // E5B Pilot
+		case 18: return CODE_L6B; // E6B
+		case 22: return CODE_L6C; // E6C
+		default: return 0;
+		}
+	}
+	else if (sys == SYS_QZS) {
+		switch (sigtype) {
+		case  0: return CODE_L1C; // L1C/A
+		case  3: return CODE_L1L; // L1C pilot
+		case  6: return CODE_L5I; // L5 Data
+		case 11: return CODE_L1S; // L1C Data
```

Comment:
Docs mentions QZSS case 9, with the same footnote as for GPS.

There is no mention of QZSS case 21 or 27 in the docs, no mention of L6 support?